### PR TITLE
[5.2] Minor cleanup for plugins JS

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

As Joomla no longer supports IE11, this PR mainly removes the need for converting a NodeList to an Array, as evergreen browser's implementation of `forEach` can iterater over a NodeList.

Also includes some other minor tweaks.

image

### Testing Instructions

Code review @dgrammatiko @Fedik

### CODE REVIEW

File: build/media_source/plg_captcha_recaptcha/js/recaptcha.es6.js [hunk 1]
```diff
@@ -8,10 +8,9 @@
   'use strict';
 
   window.JoomlainitReCaptcha2 = () => {
-    const elements = [].slice.call(document.getElementsByClassName('g-recaptcha'));
     const optionKeys = ['sitekey', 'theme', 'size', 'tabindex', 'callback', 'expired-callback', 'error-callback'];
 
-    elements.forEach((element) => {
+    document.getElementsByClassName('g-recaptcha').forEach((element) => {
```
Comment (dgrammatiko) [inline build/media_source/plg_captcha_recaptcha/js/recaptcha.es6.js]:
```suggestion
    document.querySelectorAll('.g-recaptcha').forEach((element) => {
```

Please don't use `getElementsByClassName`

File: build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js [hunk 2]
```diff
@@ -30,7 +30,5 @@ const initScheduler = () => {
 };
 
 ((document) => {
-  document.addEventListener('DOMContentLoaded', () => {
-    initScheduler();
-  });
+  document.addEventListener('DOMContentLoaded', () => initScheduler);
```
Comment (Fedik) [inline build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js]: That a bug :bug:

File: build/media_source/plg_captcha_recaptcha/js/recaptcha.es6.js (hunk 1 above)
Comment (C-Lodder) [reply build/media_source/plg_captcha_recaptcha/js/recaptcha.es6.js]: If you're sure there wasn't a reason for fetching a **live** HTMLCollection, then happy to move to static call.

File: build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js (hunk 2 above)
Comment (C-Lodder) [reply build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js]: ?

Comment (Fedik) [reply build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js]: Look closely how you define the listener here :wink:

File: build/media_source/plg_captcha_recaptcha/js/recaptcha.es6.js (hunk 1 above)
Comment (dgrammatiko) [reply build/media_source/plg_captcha_recaptcha/js/recaptcha.es6.js]: @C-Lodder if the liveliness of the method was used then this part would had to be on an event triggered from some DOM mutation. Reading the code that's not what's happening here

File: build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js (hunk 2 above)
Comment (dgrammatiko) [inline build/media_source/plg_system_schedulerunner/js/run-schedule.es6.js]:
```suggestion
  document.addEventListener('DOMContentLoaded', () => initScheduler);
```

Either:
```
document.addEventListener('DOMContentLoaded', () => initScheduler());
```

Or

```
document.addEventListener('DOMContentLoaded', initScheduler);
```

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 0a28f6e5e78cedd95b2d6021cacbab3e14789113<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42797">issues.joomla.org/tracker/joomla-cms/42797</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 0a28f6e5e78cedd95b2d6021cacbab3e14789113<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42797">issues.joomla.org/tracker/joomla-cms/42797</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42797">issues.joomla.org/tracker/joomla-cms/42797</a>.</sub>

**Comment by pe7er** [comment]
Thanks @C-Lodder !

**Pull request merged**